dag_in_context/src/lib.rs

```diff
@@ -116,11 +116,15 @@ pub fn print_with_intermediate_vars(termdag: &TermDag, term: Term) -> String {
     printed
 }
 
-// It is expected that program has context added
+// Build an egglog program that optimizes a particular batch of functions `fns`
+// with a schedule `schedule`.
+// If `inline_program` is true, it also inlines calls in `fns`.
+// `inline_program` is the program to inline calls from, allowing us to inline unoptimized function bodies.
 pub fn build_program(
     program: &TreeProgram,
+    inline_program: Option<&TreeProgram>,
```

Comment:
Do we need both program and inline_program? How are they different? Is `inline_program` the "full" program and `program` a batch of programs to be optimized?

Reply:
The inline program is the original program- if you make them the same, you might inline something that has already been inlined, causing blowup